persist/dynamo/persister.go

```diff
@@ -0,0 +1,199 @@
+package dynamo
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+const (
+	// You can't use constants in struct tags (see https://github.com/golang/go/issues/4740),
+	// so if these are ever changed, they must also be replaced in all `dynamodbav` tags.
+	pkName    = "pk"
+	skName    = "sk"
+	gsiSkName = "altSort"
+	gsiName   = "gsi_" + skName + "_" + gsiSkName
+
+	idSeparator = "#"
+)
+
+// Persister persists the model objects to DynammoDB
+type Persister struct {
+	svc       *dynamodb.DynamoDB
+	tableName *string
+}
+
+// NewPersister constructs a new Persister
+func NewPersister(session *session.Session, tableName string) (Persister, error) {
+	svc := dynamodb.New(session)
+	err := ensureTableExists(svc, tableName)
+	p := Persister{
+		svc:       svc,
+		tableName: &tableName,
+	}
+	return p, err
+}
+
+// getSequenceValue returns a unique sequence value
+func (p *Persister) getSequenceValue() (uint32, error) {
```

Comment:
I'm not sure how important it is, but this will become a hotspot if every insert requires an update to this value. What do you think about using random 53-bit integers? The collision probability would be low, roughly k^2/2^54, where k is the total number of items in the database: https://preshing.com/20110504/hash-collision-probabilities/ Or we could use say 100 sequence generators (with primary keys of 0..99) and have getSequenceValue() pick a generator at random and then multiply the sequence number by 100 and add the generator number to return a unique id.

Reply:
I just ran the numbers, and random 53-bit numbers won't work in this case since we'll have potentially millions of records.  But having 100 sequence generators would spread out the hotspot. I'm thinking when a spreadsheet with 10k rows is uploaded, it would be nice to not have a hotspot.

Reply:
The other option would be to allow grabbing a block of ids at a time. If the persister code grabs 100 at a time and keeps track of them in memory, that's only 100 calls for the 10,000 row spreadsheet. You get gaps in the sequence when you restart servers, but that's not an issue.

Reply:
That sounds good. The records writer knows how many ids it needs, so it could grab them all in one go. How about making the number of ids to return a parameter?

Reply:
If it knows how many it needs, that's even easier. I'll add a method.

Reply:
Thanks!